src/tyr/serializers.cc

```diff
@@ -23,59 +23,48 @@
 
 
 using namespace valhalla;
+using namespace valhalla::baldr;
 using namespace valhalla::tyr;
 using namespace std;
 
-namespace {
+namespace osrm {
 
-  namespace osrm_serializers {
+  valhalla::baldr::json::MapPtr waypoint(const valhalla::odin::Location& location) {
+    // Create a waypoint to add to the array
+    auto waypoint = json::map({});
 
-/** TODO - make the OSRM serializer common so it can be used by matrix, route, locate, etc.
-    // Serialize locations (called waypoints in OSRM). Waypoints are described here:
-    //     http://project-osrm.org/docs/v5.5.1/api/#waypoint-object
-    json::ArrayPtr waypoints(const std::list<valhalla::odin::TripDirections>& legs){
-      int index = 0;
-      auto waypoints = json::array({});
-      for (auto leg = legs.begin(); leg != legs.end(); ++leg) {
-        for (auto location = leg->location().begin() + index; location != leg->location().end(); ++location) {
-          index = 1;
+    // Output location as a lon,lat array. Note this is the projected
+    // lon,lat on the nearest road.
+    PointLL input_ll(location.ll().lng(), location.ll().lat());
+    PointLL proj_ll(location.projected_ll().lng(), location.projected_ll().lat());
+    auto loc = json::array({});
+    loc->emplace_back(json::fp_t{proj_ll.lng(), 6});
+    loc->emplace_back(json::fp_t{proj_ll.lat(), 6});
+    waypoint->emplace("location", loc);
 
-          // Create a waypoint to add to the array
-          auto waypoint = json::map({});
+    // Add street name.
+    waypoint->emplace("street", location.name());
```

Comment:
JSON key be "name". I tried using location.name() and location.street() but I am getting a blank string for this request:

http://localhost:8002/route?json={%22locations%22:[{%22lat%22:39.628063,%22lon%22:-76.120684},{%22lat%22:39.623649,%22lon%22:-76.129057}],%22costing%22:%22auto%22,%22directions_options%22:{%22language%22:%22en-US%22}}&format=osrm

Compare to OSRM request where name is the street name of the starting or ending edge of the path.